### PR TITLE
Fix schedule inline transform

### LINKED ISSUE #6321: Schedule Call inlining has a bug
### Information

- **Qiskit Terra version**: commit 3c979ebdacecbbd213757a8149bddc309bfa58c0
- **Python version**: 3.8.5
- **Operating system**: Ubuntu 20.04

### What is the current behavior?

Inlining the call instructions of a schedule does not remove all call instructions.

### Steps to reproduce the problem

The following code should reproduce this behavior:
```python
from qiskit.pulse.transforms import inline_subroutines
from qiskit.circuit import Parameter
import qiskit.pulse as pulse
from qiskit.pulse import Drag, DriveChannel, Gaussian


amp = Parameter("amp")
d0_ = DriveChannel(Parameter('ch0'))
sigma = Parameter("σ")

with pulse.build(name="xp") as xp:
    pulse.play(Gaussian(160, amp, sigma), d0_)

with pulse.build(name="xp2") as xp2:
    with pulse.align_sequential():
        pulse.call(xp)
        
print(xp2)
print(inline_subroutines(xp2))
```

The output is
```
ScheduleBlock(ScheduleBlock(Call(ScheduleBlock(Play(Gaussian(duration=160, amp=amp, sigma=σ), DriveChannel(ch0)), name="xp", transform=AlignLeft()), name='call_xp'), name="block2", transform=AlignSequential()), name="xp2", transform=AlignLeft())
ScheduleBlock(ScheduleBlock(Call(ScheduleBlock(Play(Gaussian(duration=160, amp=amp, sigma=σ), DriveChannel(ch0)), name="xp", transform=AlignLeft()), name='call_xp'), name="block2", transform=AlignSequential()), name="xp2", transform=AlignLeft())
```

### What is the expected behavior?

All call instructions should have been removed by the inline.

### Suggested solutions

There might be an issue with the inline function.

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
This is the high priority patch. These call instructions left in the schedule can crash job execution.

Fix #6321

### CODE REVIEW

File: qiskit/pulse/transforms/canonicalization.py [hunk 1]
```diff
@@ -157,7 +157,9 @@ def _inline_schedule(schedule: Schedule) -> Schedule:
     """
     ret_schedule = Schedule(name=schedule.name,
                             metadata=schedule.metadata)
-    for t0, inst in schedule.instructions:
+    for t0, inst in schedule._children:
```
Comment (eggerdj) [inline qiskit/pulse/transforms/canonicalization.py:160]: Any way to avoid accessing a private variable?

Comment (taalexander) [reply qiskit/pulse/transforms/canonicalization.py:160]: I don't think so, I would be fine with this becoming a public variable in a separate patch. The old pre-block builder had to use this as well.

**Review by taalexander — DISMISSED** [review]

**Comment by taalexander** [comment]
Approval pending CI passing.

**Comment by nkanazawa1989** [comment]
Sorry I changed method to public before reading your reply @taalexander. This has been private because `Schedule` and `Instruction` had been composite pattern implementation (only schedule has children). Currently this pattern is no longer used and I think changing the children to public doesn't affect any other modules.

**Comment by taalexander** [comment]
I agree, thank you.

**Review by eggerdj — APPROVED** [review]
LGTM

File: qiskit/pulse/schedule.py [hunk 2]
```diff
@@ -201,16 +201,20 @@ def channels(self) -> Tuple[Channel]:
         return tuple(self._timeslots.keys())
 
     @property
-    def _children(self) -> Tuple[Tuple[int, ScheduleComponent], ...]:
-        """Return the child``NamedValues``s of this ``Schedule`` in the
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:205]: This should instead raise a deprecation warning.

File: qiskit/pulse/schedule.py [hunk 3]
```diff
@@ -364,7 +367,7 @@ def _mutable_insert(self,
             schedule: Schedule to mutably insert.
         """
         self._add_timeslots(start_time, schedule)
-        self.__children.append((start_time, schedule))
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:367]: We will have to keep as `__children` until `_children` can be deprecated.

**Review by taalexander — CHANGES_REQUESTED** [review]
We should first give a deprecation period for `_children` others may have been using as we have been since it was necessary to get at the structured contents of the class.

**Comment by nkanazawa1989** [comment]
Thank you @taalexander . I reverted these changes and added deprecation warning.

File: qiskit/pulse/schedule.py [hunk 4]
```diff
@@ -202,9 +202,31 @@ def channels(self) -> Tuple[Channel]:
...
+        Notes:
+            Nested schedules are returned as-is. If you want to collect only instructions,
+            use py:meth:`~Schedule.instructions` instead.
+
+        Returns:
+            A tuple, where each element is a two-tuple containing the initial
+            scheduled time of each ``NamedValue`` and the component
+            itself.
+        """
+        warnings.warn('Schedule._children is now public method Schedule.children. '
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:217]:
```suggestion
        warnings.warn('Schedule._children is now available as the public property Schedule.children. '
```

File: qiskit/pulse/schedule.py [hunk 5]
```diff
@@ -202,9 +202,31 @@ def channels(self) -> Tuple[Channel]:
...
+                      'Access to the private member is being deprecated.', DeprecationWarning)
```
Comment (taalexander) [inline qiskit/pulse/schedule.py:218]:
```suggestion
                      'Access to this private property is being deprecated.', DeprecationWarning)
```

**Review by taalexander — CHANGES_REQUESTED** [review]

**Review by taalexander — DISMISSED** [review]

**Review by taalexander — APPROVED** [review]

**Pull request merged**